src/cast_core/src/CollectionScanner.cpp

```diff
@@ -219,15 +234,22 @@ void collectionScan(CollectionScanner::PhaseConfig* config,
     bool scanFinished = false;
     auto statTracker = config->scanOperation.start();
     for (auto& collection : collections) {
-        auto filter = config->filterExpr ? config->filterExpr->evaluate()
-                                         : bsoncxx::document::view_or_value{};
-        auto docs = collection.find(session, filter, config->findOptions);
+        auto docs = [&]() {
+          if (config->aggregate) {
+              return collection.aggregate(session, evaluatePipeline(config), config->aggregateOptions);
```

Comment:
Should this really be a return? Wont' that execute once and then end as oppose to repeating in the for loop?

Reply:
I'm not sure I get what you mean. 
There is a lambda for each iteration of the outer _for (auto& collection : collections) {_  and each lambda returns a cursor so the inner _for (auto& doc : docs()) {_ works as expected.

Reply:
My eyes missed the lambda. Could you add a comment for me?

Reply:
sure, also renamed docs to cursor.